lib/ansible/modules/network/cloudengine/ce_switchport.py

```diff
@@ -0,0 +1,826 @@
+            if int(self.native_vlan) <= 0 or int(self.native_vlan) > 4094:
+                self.module.fail_json(
+                    msg='Error: Native vlan id is not in the range from 1 to 4094.')
+
+        # check mode
+        if self.state == "present" or self.state == "absent":
+            if not self.mode:
+                self.module.fail_json(msg='Error: Interface mode must be set.')
+
+        # get interface info
+        self.intf_info = self.get_interface_dict(self.interface)
+        if not self.intf_info:
+            self.module.fail_json(msg='Error: Interface does not exists.')
+
+        if not self.is_l2switchport():
+            self.module.fail_json(
+                msg='Error: Interface is not layer2 swtich port.')
+
+    def get_proposed(self):
+        """get proposed info"""
+
+        self.proposed['state'] = self.state
+        self.proposed['interface'] = self.interface
+        self.proposed['mode'] = self.mode
+        self.proposed['access_vlan'] = self.access_vlan
+        self.proposed['native_vlan'] = self.native_vlan
+        self.proposed['trunk_vlans'] = self.trunk_vlans
+
+    def get_existing(self):
+        """get existing info"""
+
+        if self.intf_info:
+            self.existing["interface"] = self.intf_info["ifName"]
+            self.existing["mode"] = self.intf_info["linkType"]
+            self.existing["switchport"] = self.intf_info["l2Enable"]
+            self.existing['access_vlan'] = self.intf_info["pvid"]
+            self.existing['native_vlan'] = self.intf_info["pvid"]
+            self.existing['trunk_vlans'] = self.intf_info["trunkVlans"]
+
+    def get_end_state(self):
+        """get end state info"""
+
+        if self.intf_info:
+            end_info = self.get_interface_dict(self.interface)
+            if end_info:
+                self.end_state["interface"] = end_info["ifName"]
+                self.end_state["mode"] = end_info["linkType"]
+                self.end_state["switchport"] = end_info["l2Enable"]
+                self.end_state['access_vlan'] = end_info["pvid"]
+                self.end_state['native_vlan'] = end_info["pvid"]
+                self.end_state['trunk_vlans'] = end_info["trunkVlans"]
+
+    def work(self):
+        """worker"""
+
+        self.check_params()
+        if not self.intf_info:
+            self.module.fail_json(msg='Error: interface does not exists.')
+
+        self.get_existing()
+        self.get_proposed()
+
+        # present or absent
+        if self.state == "present" or self.state == "absent":
+            if self.mode == "access":
+                self.merge_access_vlan(self.interface, self.access_vlan)
+            elif self.mode == "trunk":
+                self.merge_trunk_vlan(
+                    self.interface, self.native_vlan, self.trunk_vlans)
+            else:
+                self.module.fail_json(msg='Error: Interface mode must be set.')
```

Comment:
There are some options to `AnsibleModule` which you may be able to use to validate options passed in.
Look at existing modules for examples:
```
mutually_exclusive
required_together
required_one_of
require_if
```